support/vs-code/silverlsp/src/extension.ts

```diff
@@ -25,15 +26,26 @@ export function activate(context: vscode.ExtensionContext) {
 		let excecutable: string = path.join(JAVA_HOME, 'bin', 'java');
 
 		// path to the launcher.jar
+		let classPath = [];
+
+		let compilerJars: string = config.get('compilerJar') || "";
 		let launcherJar: string = path.join(__dirname, '..', 'launcher', 'launcher.jar');
-		let compilerJar: string = config.get('compilerJar') || "";
-		if (compilerJar && !compilerJar.startsWith("/") && vscode.workspace.workspaceFolders !== undefined) {
-			compilerJar = path.join(vscode.workspace.workspaceFolders[0].uri.path, "/", compilerJar);
+		for (let compilerJar of compilerJars.split(':')) {
+			if (!compilerJar.startsWith("/") && vscode.workspace.workspaceFolders !== undefined) {
+				compilerJar = path.join(vscode.workspace.workspaceFolders[0].uri.path, "/", compilerJar);
+			}
+			classPath.push(compilerJar);
 		}
-		let classPath = [launcherJar];
-		if (compilerJar) {
-			classPath.unshift(compilerJar);
+		classPath.push(launcherJar);
+
+		for (let jar of classPath) {
+			// Check if the jar file exists, and if not, raise an error.
+			if (!fs.existsSync(jar)) {
```

Comment:
should activate be an async function, and should this be the version from `fs/promises`? https://code.visualstudio.com/api/references/vscode-api makes it sound like `Extension<T>.activate` should return a `Thenable<T>`

Reply:
No idea, I copied this from a tutorial a long time ago. Added a TODO comment for now.